### ISSUE TITLE
Does not work on M1 Mac.

### ISSUE THREAD

**Issue by 32Bites** [tehzz/homebrew-n64-dev#2]
This fails to compile on an M1 Mac as it seems by default it requires it to be amd64 to build. The formula: mips64-elf-gcc.

**Comment by tehzz** [comment]
I don't think that GCC officially supports M1/arm macs yet, though I see that homebrew [has an unofficial build in its official gcc formula](https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/gcc.rb). I don't have an apple silicon mac to check if the `mips64-elf` target builds...

**Comment by CrepeGoat** [comment]
I'm on an M1 mac; `mips-elf-binutils` seems to build fine, `mips64-elf-gcc` seems to fail.

**Comment by Ibxx0403** [comment]
I also have some issues as i cannot  seem to build mips64-elf-binutils I have attached the error message down below. 

checking whether strnlen is declared... yes
checking whether canonicalize_file_name must be declared... yes
checking for stdlib.h... (cached) yes
checking for unistd.h... (cached) yes
checking for sys/param.h... (cached) yes
checking for getpagesize... (cached) yes
checking for working mmap... yes
checking for working strncmp... no
configure: updating cache ./config.cache
configure: creating ./config.status
config.status: creating Makefile
config.status: creating testsuite/Makefile
config.status: creating config.h
config.status: executing default commands
make: *** [all] Error 2

**Comment by wgreenberg** [comment]
i had the same error as @Ibxx0403 on an M4 mac

**Comment by enumouse** [comment]
Can confirm I get the same issue here, also on an M4 mac. I recall getting a successful build using another provider for these binaries back on my M1, but I cannot recall exactly where I got them from which is beyond annoying.

If I have the time I will have a look around for any sign of what I used before, or attempt a fix for these formulae. My main concern is being much more familiar with Clang than GCC, but I'll see what I can do.

```
==> make
Last 15 lines from /Users/calamari/Library/Logs/Homebrew/mips64-elf-binutils/02.make:
checking whether strnlen is declared... yes
checking whether canonicalize_file_name must be declared... yes
checking for stdlib.h... (cached) yes
checking for unistd.h... (cached) yes
checking for sys/param.h... (cached) yes
checking for getpagesize... (cached) yes
checking for working mmap... yes
checking for working strncmp... no
configure: updating cache ./config.cache
configure: creating ./config.status
config.status: creating Makefile
config.status: creating testsuite/Makefile
config.status: creating config.h
config.status: executing default commands
make: *** [all] Error 2
```

**Comment by wndalci** [comment]
**Fix/Workaround!** While I can confirm this is unfortunately broken, it is possible to use [Libdragon](https://github.com/DragonMinded/libdragon)'s toolchain to compile a byte-matching rom! I used this instead of these formula, and have gotten successful builds
```
Building ROM: build/eu/sm64.eu.elf -> build/eu/sm64.eu.z64
Checking if ROM matches..
sm64.eu: OK
```

### Full Instructions (Adapted from Libdragon Wiki for this usage)
- Make sure you have set a path for Libdragon in your shell config: `export N64_INST=/opt/libdragon`. Relaunch/Source your terminal if you have made changes.
- Clone the [Libdragon repository](https://github.com/DragonMinded/libdragon) and `cd` into it.
- Navigate into the `tools` directory and run `chmod +x ./build-toolchain.sh`, then `./build-toolchain.sh`
- Make yourself some tea, just keep in mind it may ask for sudo password.
- Add the binaries to path from within your shell config, such as `export PATH="/opt/libdragon/bin:$PATH"`. Remember to relaunch your terminal!
- You can now build as usual! In my case, I had to fix a common build error with [this pull request's patch.](https://github.com/n64decomp/sm64/pull/77)

For full Libdragon installation guide, [see here](https://github.com/DragonMinded/libdragon/wiki/Installing-libdragon). Hope this helps :)
